doc/changes/fixed/12834.md

```diff
@@ -0,0 +1,2 @@
+- Don’t hardcode compiler packages when determining the compiler for developer
```

Comment:
We usually don't write changelog entries for `dune pkg` related changes.

I think it would be useful to revisit the policy but this is how it is at the moment…